source/common/router/router.h

```diff
@@ -477,12 +477,9 @@ class Filter : Logger::Loggable<Logger::Id::router>,
                    Runtime::Loader& runtime, Runtime::RandomGenerator& random,
                    Event::Dispatcher& dispatcher, Upstream::ResourcePriority priority) PURE;
 
-  using HttpOrTcpPool =
-      absl::variant<Http::ConnectionPool::Instance*, Tcp::ConnectionPool::Instance*>;
-  HttpOrTcpPool createConnPool(Upstream::HostDescriptionConstSharedPtr& host);
-  UpstreamRequestPtr createUpstreamRequest(Filter::HttpOrTcpPool conn_pool);
+  std::unique_ptr<GenericConnPool> createConnPool();
+  UpstreamRequestPtr createUpstreamRequest(std::unique_ptr<GenericConnPool>&& conn_pool);
```

Comment:
Mark: I don't see the definition in PR... github issue?

Reply:
I didn't find the impl anywhere but envoy is built successfully... Please educate me
My initial intention is to argue if we need the rvalue reference "&&" in parameter